presidio_evaluator/experiment_tracking/local_tracker.py

```diff
@@ -0,0 +1,103 @@
+            Each new experiment generate an experiment directory and stores the following files inside the directory:
+            1. confusion_matrix.csv - The generated confusion matrix for each entity
+            2. experiment_params.json - Dictionary containing any logged information needed to the user
+            3. experiment_errors.pkl - Pickles representation of a `List[ModelError]` instances. 
+
+        Args:
+            directory (str): Main location of experimentation folder
+            experiment_name (str): Name of experiment to store information
+        """
+        super().__init__()
+        self.experiment_name = experiment_name
+        self.dir = directory
+        self.start()
+
+    def start(self):
+        """Generates a new folder within the experiments directory 
+        """
+        full_path = Path(Path.cwd(), self.dir, self.experiment_name)
+        full_path.mkdir(parents=True, exist_ok=True)
+
+    def log_errors(self, errors: List[ModelError]):
+        """ Generate a serialized version of all model errors and store as `.pkl` file
+        Args:
+            errors (List[ModelError]): List containing generated errors by the NER model
+        """
+        errors_str = self._serialize_errors(errors)
+        file_path = Path(Path.cwd(), self.dir, self.experiment_name,
+                         "experiment_errors.pkl")
+        file = open(file_path, 'wb')
+        pickle.dump(errors_str, file)
+        file.close()
+
+    def _serialize_errors(self, errors: List) -> List[dict]:
+        errors_str = list()
+        for error in errors:
+            error.vector = error.token.vector
+            error.token = str(error.token)
+            errors_str.append(error)
+        return errors_str
+
+    def log_confusion_matrix_table(
+        self,
+        matrix: List[List[int]],
+        labels=List[str],
+    ):
+        """Store confusion matrix as a .csv file
+
+        Args:
+            matrix (List[List[int]]): Error counts
+            labels (_type_, optional): Names of entities. Defaults to List[str].
+        """
+        self.confusion_matrix = matrix
+        self.labels = labels
+        file_path = Path(Path.cwd(), self.dir, self.experiment_name,
+                         "confusion_matrix.csv")
+        # store confusion matrix
+        pd.DataFrame(matrix, columns=labels, index=labels).to_csv(file_path)
+
+    def end(self):
+        """Store all logged information into a JSON file
+        """
+        file_path = Path(Path.cwd(), self.dir, self.experiment_name,
+                         "experiment_params.json")
+        print(f"saving experiment data to directory {self.experiment_name}")
+        with file_path.open('w') as json_file:
+            json.dump(self.__dict__, json_file)
+
+
+if __name__ == "__main__":
+
+    directory = Path(Path.cwd(), 'experiments')
+    experiment_name = "my_first_experiment"
```

Comment:
The code looks great to me. Just a minor comment. Should we add the datetime in the experiment name, so the user can track when the experiment is done?

Reply:
It's a great idea, I will add it as a parameter in the `.json` file, this way the experiment_name will remain clear and easy to remember but the the date time of the experiment execution will be stored. What do you think?

Reply:
I think that would be a great option. Just a question, if a user runs the same experiment name but in 2 different timestamps, the old results will be overwritten by the new one?

Reply:
At the moment yes.

Reply:
Ok. In this case, I think we add datetime in the .json file.

Reply:
done

Reply:
Great! Thank Mel!